packages/core/src/utils/ui-helpers.ts

```diff
@@ -0,0 +1,20 @@
+export async function waitUntilElementVisible(
+  id: string,
+  timeout = 1000,
+  step = 100
+): Promise<boolean> {
+  let counter = 0;
+
+  return new Promise(resolve => {
```

Comment:
[MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) might be a cleaner solution, but let's test and verify this is the fix and we can improve afterwards.